[bug] Receiver does not block after consuming last item; pops empty deque.

### Issue desciption:
In ThreadSafeStreamReceiver, when the last item is consumed and the queue becomes empty, the added event remains set.
The next next(receiver) call proceeds without waiting and attempts popleft() on an empty deque, causing an IndexError or incorrect wakeups.

### Repro Steps:
- Use the following test function `test_receiver_waits_after_queue_empty_until_new_item_sync`.
- Producer appends "A", sleeps briefly, then appends "B" and commits.
- Consumer reads two items with next(receiver) twice.
- issue: consumer hits IndexError on the second next().

```text
    def test_receiver_waits_after_queue_empty_until_new_item_sync():
        sender, receiver = create_thread_safe_stream(timeout=1.0)
        consumed: list[str] = []
    
        def producer():
            with sender:
                sender.append("A")  # queue has one item; not completed yet
                time.sleep(0.1)  # ensure consumer attempts the next() on empty queue
                sender.append("B")
                sender.commit()
    
        def consumer():
            with receiver:
                a = next(receiver)
                consumed.append(a)
                b = next(receiver)
                consumed.append(b)
    
        t1 = threading.Thread(target=producer)
        t2 = threading.Thread(target=consumer)
        t1.start()
        t2.start()
        t1.join()
        t2.join()
    
>       assert consumed == ["A", "B"]
E       AssertionError: assert ['A'] == ['A', 'B']

......

      item = self._queue.popleft()
             ^^^^^^^^^^^^^^^^^^^^^
  IndexError: pop from an empty deque

```

--- a/src/ghoshell_moss/core/helpers/stream.py
+++ b/src/ghoshell_moss/core/helpers/stream.py
@@ -68,6 +68,8 @@ class ThreadSafeStreamReceiver(Generic[ItemT]):
     def __next__(self) -> ItemT:
         if len(self._queue) > 0:
             item = self._queue.popleft()
+            if len(self._queue) == 0:
+                self._added.clear()
             if isinstance(item, Exception):
                 raise item
             elif item is None:
@@ -104,6 +106,8 @@ class ThreadSafeStreamReceiver(Generic[ItemT]):
     async def __anext__(self) -> ItemT:
         if len(self._queue) > 0:
             item = self._queue.popleft()
+            if len(self._queue) == 0:
+                self._added.clear()
             if isinstance(item, Exception):
                 raise item
             elif item is None:

--- a/tests/helpers/test_stream.py
+++ b/tests/helpers/test_stream.py
@@ -1,5 +1,6 @@
 import asyncio
 import threading
+import time
 
 from ghoshell_moss.core.helpers.stream import (
     create_thread_safe_stream,
@@ -62,3 +63,31 @@ def test_thread_send_and_receive():
     t1.join()
     t2.join()
     assert content == done[0]
+
+
+def test_receiver_waits_after_queue_empty_until_new_item_sync():
+    sender, receiver = create_thread_safe_stream(timeout=1.0)
+    consumed: list[str] = []
+
+    def producer():
+        with sender:
+            sender.append("A")  # queue has one item; not completed yet
+            time.sleep(0.1)  # ensure consumer attempts the next() on empty queue
+            sender.append("B")
+            sender.commit()
+
+    def consumer():
+        with receiver:
+            a = next(receiver)
+            consumed.append(a)
+            b = next(receiver)
+            consumed.append(b)
+
+    t1 = threading.Thread(target=producer)
+    t2 = threading.Thread(target=consumer)
+    t1.start()
+    t2.start()
+    t1.join()
+    t2.join()
+
+    assert consumed == ["A", "B"]

